lib/pbench/test/unit/server/database/test_audit_db.py

```diff
@@ -155,44 +161,69 @@ def test_queries(self, fake_db):
             )
 
             f.tick(delta=datetime.timedelta(hours=5))
-            Audit.create(root=root1, status=AuditStatus.FAILURE, attributes=attr1)
+            done1 = Audit.create(
+                root=root1, status=AuditStatus.FAILURE, attributes=attr1
+            )
 
             f.tick(delta=datetime.timedelta(minutes=40))
-            Audit.create(root=root2, status=AuditStatus.SUCCESS, attributes=attr2)
+            done2 = Audit.create(
+                root=root2, status=AuditStatus.SUCCESS, attributes=attr2
+            )
 
         # Try some queries. The mocked query returns the ordered list of
         # filters passed to the database engine: we want to confirm that the
         # expected filters were run, and there's no point in trying to mock the
         # behavior of those filters.
+        expected_commits = [
+            FakeRow.clone(root1),
+            FakeRow.clone(done1),
+            FakeRow.clone(root2),
+            FakeRow.clone(done2),
+        ]
 
         Audit.query(user_name="imme")
-        self.session.check_session(filters=["user_name=imme"])
+        self.session.check_session(
+            queries=1, committed=expected_commits, filters=["user_name=imme"]
+        )
 
         Audit.query(object_name="testy")
-        self.session.check_session(filters=["object_name=testy"])
+
+        self.session.check_session(
+            queries=1, committed=expected_commits, filters=["object_name=testy"]
+        )
 
         Audit.query(dataset=ds1)
         self.session.check_session(
-            filters=["audit.object_type = AuditType.DATASET, audit.object_id = md5.1"]
+            queries=1,
+            committed=None,
```

Comment:
Not sure why you opted to specify `committed=None` for this call, unlike all the calls around it, but...OK.  😀

Reply:
I suppose just for variety, since I'd already validated the commit list multiple times. I suppose I could have done it just once and used `None` for the rest. It was totally arbitrary.